Sources/InfomaniakCore/ItemProviderRepresentation/ItemProviderWeblocRepresentation.swift

```diff
@@ -65,12 +65,21 @@ public final class ItemProviderWeblocRepresentation: NSObject, ProgressResultabl
 
             do {
                 @InjectService var pathProvider: AppGroupPathProvidable
+
+                // Use a unique folder to prevent collisions
                 let tmpDirectoryURL = pathProvider.tmpDirectoryURL
                     .appendingPathComponent(UUID().uuidString, isDirectory: true)
                 try fileManager.createDirectory(at: tmpDirectoryURL, withIntermediateDirectories: true)
 
-                let fileName = path.deletingPathExtension().lastPathComponent
-                let targetURL = tmpDirectoryURL.appendingPathComponent("\(fileName).webloc")
+                let currentName = (path.lastPathComponent as NSString).deletingPathExtension
```

Comment:
it is not equivalent.

One starts with the lastPathComponent, and remove the extention, checks if empty.

What you mean (not what is written) probably is 
> path.deletingPathExtension().lastPathComponent

This one is a bug, as if the file name is empty, it will catch the previous component (ex. folder name) instead.